apps/users-service/src/address-book-service.service.ts

```diff
@@ -107,4 +113,33 @@ export class AddressBookService {
 
     return { status: 1 }
   }
+
+  async getAddressByPin (userId: string, pin: number): Promise<PinAddressI> {
+    try {
+      const getUser: UserI = await this.usersRepository.findOne({
+        _id: userId,
+        addressPin: pin
+      })
+
+      const addresses: AddressBookI[] = await this.repository.findAndPopulate(
+        {
+          userId: getUser._id.toString(),
+          isDeleted: false
+        },
+        ['labelId']
+      )
+
+      return {
+        user: getUser,
```

Comment:
No need to return the user in the PinAddressI. firstName, LastName, and address: AddressI (only adresses where sharable is true)